pandas/core/internals/managers.py

```diff
@@ -1331,6 +1331,10 @@ def _slice_take_blocks_ax0(self, slice_or_indexer, fill_tuple=None):
             blk = self.blocks[0]
 
             if sl_type in ("slice", "mask"):
+                if blk.is_extension:
+                    # GH#32959 EABlock would fail since we cant make 0-width
+                    if sllen == 0:
+                        return []
```

Comment:
Is `sllen` the length of the indexer or the length of the slice?

I would think that you could do a slice like `ExtensionArray[:0]` with a length of 0, and return a length-0 extension block / array.

Reply:
sllen is the number of columns we expect to get back, and we cant represent a 0-width EA

Reply:
So the result is something like `DataFrame(index=...)`?

Is there any benefit to restricting this check to just `blk.is_extension`, or can we make this check apply to all block types since we're returning an empty result anyway?

Reply:
without the is_extension check i got test failures

Reply:
Can you share them? Or am I completely off base for thinking that EAs shouldn't (need to be) treated specially here?

Reply:
naturally, re-running today im no longer getting any failures when doing this unconditionally

Reply:
removed this check